src/test/java/com/amazon/ion/system/IonWriterBuilderTestBase.java

```diff
@@ -0,0 +1,169 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+package com.amazon.ion.system;
+
+import com.amazon.ion.IonCatalog;
+import com.amazon.ion.IonWriter;
+import com.amazon.ion.SymbolTable;
+import com.amazon.ion.impl.Symtabs;
+import com.amazon.ion.impl._Private_IonWriter;
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.io.ByteArrayOutputStream;
+import java.io.OutputStream;
+
+import static org.junit.Assert.assertArrayEquals;
+import static org.junit.Assert.assertNotSame;
+import static org.junit.Assert.assertSame;
+import static org.junit.Assert.fail;
+
+abstract class IonWriterBuilderTestBase<Builder extends IonWriterBuilderBase<Builder>> {
```

Comment:
I think it's interesting that you're intentionally splitting the Ion 1.0 and Ion 1.1 builders from each other, but you're unifying the tests. Doesn't need to be done in this PR, but it might be good to have some explanatory comments on this test class about what can and cannot be tested here.

Reply:
I'm splitting them at the interface level, but in the implementation they still share a common ancestor: `IonWriterBuilderBase`. I'll add a comment that this class tests builders that inherit from `IonWriterBuilderBase`.